### PR TITLE
Sign in and sign up

### PR DESCRIPTION
**Pull request by rodrigob1991**

### CODE REVIEW

File: packages/mobile/screens/SignIn.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                                    secureTextEntry
+                                    placeholder="Password..."
+                                    placeholderTextColor="#003f5c"
+                                    onChangeText={handleChange('password')}
+                                    error={touched["password"] && errors["password"]?.length > 0}
+                                    onBlur={handleBlur('password')}
+                                />
+                            </Input>
+
+                            <ForgotText >{'Forgot de password?'}</ForgotText>
```
Comment (DogChef) [inline packages/mobile/screens/SignIn.tsx:101]: the*

In the design it's "Forgot password?" so please  double check the designs

File: packages/mobile/screens/SignIn.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+                            {errors?.server &&
+                                <ErrorText>{errors.server}</ErrorText>
+                            } 
+                            <SignInButton
+                                onPress={handleSubmit}>
+                                <SignInText>SIGN IN</SignInText>
```
Comment (DogChef) [inline packages/mobile/screens/SignIn.tsx:108]: Also, it's Login, not Sign in, please check the [designs](https://www.figma.com/file/AQcZJIrJXkb2RnMaHyMz90/Material-Design-Theme-Kit?node-id=19537%3A72), they're there for a reason

File: packages/mobile/screens/SignIn.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                            </SignInButton>
+                        </React.Fragment>
+
+
+                    )}
+            </Formik>
+            <SignUpFrame>
+                <Text>{'Do not have an acount?'}</Text>
+                <TouchableOpacity onPress={() => { navigation.navigate("SignUp") }}>
+                    <SignUpText> SIGN UP</SignUpText>
```
Comment (DogChef) [inline packages/mobile/screens/SignIn.tsx:118]: Unnecessary space before text

File: packages/mobile/screens/SignIn.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+        </Container>
+    );
+}
+
+const Header = styled.View`
+  margin-bottom: 25px;
+  `;
+
+const TextInputCust = styled(TextInput)`
```
Comment (DogChef) [inline packages/mobile/screens/SignIn.tsx:130]: What does Cust mean?

File: packages/mobile/screens/SignIn.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    color: black;
+    font-size: 10px;
+
+`;
+
+const WelcomeText = styled.Text`
+    color: #ffb900
+`;
+
+const SignUpText = styled.Text`
```
Comment (DogChef) [inline packages/mobile/screens/SignIn.tsx:195]: Isn't this the same as welcome text? Please try to reuse your components

File: packages/mobile/screens/SignUp.tsx [hunk 6]
```diff
@@ -0,0 +1,244 @@
+import { useMutation } from "@apollo/react-hooks";
+import { Field, Formik } from 'formik';
+import gql from "graphql-tag";
+import * as React from "react";
+import { Image, Text, TouchableOpacity, Alert} from 'react-native';
+import styled from 'styled-components/native';
+import * as Yup from "yup";
+import { TextInput } from "react-native-paper";
+import CREATE_USER_MUTATION from '../local_core/mutations/CREATE_USER_MUTATION';
```
Comment (DogChef) [inline packages/mobile/screens/SignUp.tsx:9]: View the doc on how to do a proper mutation

File: packages/mobile/screens/SignUp.tsx [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+  email: Yup.string()
+    .email("Please enter a valid email")
+    .required("Please enter an email"),
+  password: Yup.string().required("Please enter the password"),
+  confirmation: Yup.string()
+    .required("Please enter the password confirmation")
+    .oneOf([Yup.ref("password")], "Password does not match")
+});
+
+function alert(nombre : string)  {Alert.alert(
```
Comment (DogChef) [inline packages/mobile/screens/SignUp.tsx:25]: What is this function for?, avoid commiting unnecesary code

File: packages/mobile/package.json [hunk 8]
```diff
@@ -19,6 +19,9 @@
     "@react-navigation/drawer": "^5.3.4",
     "@react-navigation/native": "^5.1.3",
     "@react-navigation/stack": "^5.2.6",
+    "@types/styled-components": "^5.0.1",
```
Comment (DogChef) [inline packages/mobile/package.json:22]: This should be a devDependency since its a type, right?

File: packages/mobile/package.json [hunk 9]
```diff
@@ -19,6 +19,9 @@
...
+    "@types/yup": "^0.26.34",
+    "@workspace-library/core": "1.0.0",
```
Comment (DogChef) [inline packages/mobile/package.json:24]: We're not using core as a package anymore, please remove this.

File: packages/mobile/screens/SignUp.tsx [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+      </SignInFrame>
+    </Container>
+  );
+}
+
+const Header = styled.View`
+  margin-bottom: 25px;
+  `;
+ 
+const TextInputCust = styled(TextInput)`
```
Comment (DogChef) [inline packages/mobile/screens/SignUp.tsx:172]: Seems like you're reusing a lot of components on both views, why not export them in a view and import them on the other one if they're the same one?

File: packages/mobile/screens/SignIn.tsx [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+
+const LoginSchema = Yup.object().shape({
+    email: Yup.string()
+        .email("Please enter a valid email")
+        .required("Please enter an email"),
+    password: Yup.string().required("Required")
+});
+
+
+export default function SignUp({ navigation }: { navigation: any }) {
```
Comment (DogChef) [inline packages/mobile/screens/SignIn.tsx:19]: Please define the function using arrow functions and then use export default with it, it improves readability

File: packages/mobile/screens/SignUp.tsx [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+
+  return (
+    <Container>
+      <Header>
+        <Image resizeMode={'cover'}
+          source={require('../assets/logo-lithium.png')}
+        />
+        <WelcomeText>{'Welcome to Lithium KB'}</WelcomeText>
+      </Header>
+      <Title>{'Sign up'}</Title>
```
Comment (DogChef) [inline packages/mobile/screens/SignUp.tsx:78]: You can do this with <Title>Sign up</Title>

File: packages/mobile/screens/SignUp.tsx [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+
+            {touched.confirmation && errors?.confirmation?.length &&
+              <Text style={{ fontSize: 10, color: 'red' }}>{errors.confirmation}</Text>
+            }
+            <Input>
+              <Field
+                component={TextInputCust}
+                value={confirmation}
+                secureTextEntry
+                placeholder="Confirmation..."
```
Comment (DogChef) [inline packages/mobile/screens/SignUp.tsx:139]: This isnt the appropriate placeholder

File: packages/mobile/package.json [hunk 14]
```diff
@@ -19,6 +19,9 @@
     "@react-navigation/drawer": "^5.3.4",
     "@react-navigation/native": "^5.1.3",
     "@react-navigation/stack": "^5.2.6",
+    "@types/styled-components": "^5.0.1",
+    "@types/yup": "^0.26.34",
```
[the remark this answers is not in the document]
Comment (rodrigob1991) [reply packages/mobile/package.json]: ready

File: packages/mobile/screens/SignIn.tsx [hunk 15]
```diff
@@ -0,0 +1,203 @@
+import { useMutation } from "@apollo/react-hooks";
+import { Field, Formik } from 'formik';
+import gql from "graphql-tag";
+import * as React from "react";
+import { Image, Text, TouchableOpacity, View } from 'react-native';
+import { TextInput } from "react-native-paper";
+import styled from 'styled-components/native';
+import * as Yup from "yup";
+import LOG_IN_MUTATION from '../local_core/mutations/LOG_IN_MUTATION';
```
[the remark this answers is not in the document]
Comment (rodrigob1991) [reply packages/mobile/screens/SignIn.tsx]: ready

File: packages/mobile/screens/SignIn.tsx [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+                password: values.password
+            }
+        })
+            .then(
+                ({
+                    data: {
+                        loggedUser: { id }
+                    }
+                }) => {
+                    navigation.navigate("HomeScreen")
```
[the remark this answers is not in the document]
Comment (rodrigob1991) [reply packages/mobile/screens/SignIn.tsx]: that is in another branch

File: packages/mobile/package.json (hunk 9 above)
Comment (rodrigob1991) [reply packages/mobile/package.json]: ready

File: packages/mobile/screens/SignUp.tsx (hunk 10 above)
Comment (rodrigob1991) [reply packages/mobile/screens/SignUp.tsx:154]: later

File: packages/mobile/screens/SignUp.tsx (hunk 13 above)
Comment (rodrigob1991) [reply packages/mobile/screens/SignUp.tsx:121]: ????

File: packages/mobile/screens/SignUp.tsx (hunk 12 above)
Comment (rodrigob1991) [reply packages/mobile/screens/SignUp.tsx:60]: ready

File: packages/mobile/screens/SignIn.tsx (hunk 11 above)
Comment (rodrigob1991) [reply packages/mobile/screens/SignIn.tsx:16]: improve readibility?????

File: packages/mobile/screens/SignIn.tsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (rodrigob1991) [reply packages/mobile/screens/SignIn.tsx]: reay.

File: packages/mobile/screens/SignUp.tsx (hunk 10 above)
Comment (rodrigob1991) [reply packages/mobile/screens/SignUp.tsx:175]: is only copy paste. And could have advantage not reusing.

File: packages/mobile/screens/SignUp.tsx (hunk 13 above)
Comment (rodrigob1991) [reply packages/mobile/screens/SignUp.tsx:145]: if you comment that puts the right placeholder at least

File: packages/mobile/screens/SignIn.tsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (DogChef) [reply packages/mobile/screens/SignIn.tsx]: I don't see how it was solved

File: packages/mobile/screens/SignIn.tsx [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+                <WelcomeText>{'Welcome to Lithium KB'}</WelcomeText>
+            </Header>
+            <Title>Login</Title>
+            <Formik
+                initialValues={{ email: '', password: '' }}
+                validationSchema={LoginSchema}
+                onSubmit={submition}
+            >
+                {({ values: { email, password }, handleChange, handleSubmit, errors, touched, handleBlur, }:
+                    { handleSubmit: any, values: any, handleChange: any, errors: any, touched: any, handleBlur: any }) => (
```
[the remark this answers is not in the document]
Comment (DogChef) [reply packages/mobile/screens/SignIn.tsx]: What I meant with this is: define an interface object outside the component that specifies the types. In this example it'd be:

```
interface FormikProps = {
  handleSubmit: (e?: React.FormEvent<HTMLFormElement>) => void;
  values: FormikValues;
  handleChange: (f: string) => void;
  errors: FormikErrors<FormikValues>;
  touched: FormikTouched<FormikValues>;
  handleBlur: (f: string) => void;
}

...

{
   values: { email, password },
   handleChange,
   handleSubmit,
   errors,
   touched,
   handleBlur,
}: FormikProps
```

File: packages/mobile/screens/SignUp.tsx (hunk 13 above)
Comment (DogChef) [reply packages/mobile/screens/SignUp.tsx:145]: As I already [stated before](https://github.com/Lithium-Software/workspace-library/pull/15#discussion_r408911903), please check the designs

File: packages/mobile/screens/SignIn.tsx (hunk 11 above)
Comment (DogChef) [reply packages/mobile/screens/SignIn.tsx]: Yes, what I mean with that is that it's easier to understand the code. Also it seems that the "SignIn" screen is exporting a "SignUp" function?

Comment (rodrigob1991) [reply packages/mobile/screens/SignIn.tsx]: i already changed name

File: packages/mobile/screens/SignIn.tsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (rodrigob1991) [reply packages/mobile/screens/SignIn.tsx]: i will do it in JWTAuth branch

**Pull request merged**